src/app/features/caseVisits/components/organisms/Form/Fields.tsx

```diff
@@ -0,0 +1,24 @@
+import React from "react"
+import { useForm } from "react-final-form"
+import createScaffold from "./scaffold"
+import ScaffoldFields from "app/features/shared/components/molecules/Form/ScaffoldFields"
+import type { CaseVisit } from "../../../hooks/useCaseVisits"
+
+type Props = {
+  initialValues?: CaseVisit
+}
+
+const Fields: React.FC<Props> = ({ initialValues }) => {
```

Comment:
This part probably needs a review the most

Reply:
I think its better to create a custom form-component: "AutoFillButton" or something.

```
type Props = React.ComponentProps<typeof Button> & {
  field: string
  target: string
}

const AutoFillButton:React.FC<Props> = ({ field, target, ...restProps }) => {
  const [ isDisabled, setIsDisabled ] = useState(false)
  const { getFieldState, change } = useForm()

  const handleClick = useCallback(() => {
     
    const value = [
      getFieldState(target)?.value ?? "", 
      getFieldState(field)?.value ?? ""
    ].filter(_ => _ !== "").join("\n")
    
    change("note", value)
    setIsDisabled(true)
    
  }, [ getFieldState, change, field, target ])

  return <Button {...restProps} onClick={handleClick} disabled={isDisabled}>
}
```

Register it in: 
`src/app/features/shared/components/molecules/Form/ScaffoldField.tsx`

And call it in createScaffold as:
```
note0Button: {
      type: "AutoFillButton",
      props: {
        label: "Voeg toe",
        align: "right",
        field: "note0",
        target: "note"
      }
    }
```

That way we have (unit testable) reusable component we could use elsewhere as well.

Reply:
Also: 
I think we should be careful passing functions as props without wrapping them in a `useCallback`

Reply:
Looks cleaner, yes